code/datums/components/eye/_eye.dm

```diff
@@ -0,0 +1,84 @@
+/datum/component/eye
+	///The actual eye mob linked to the component
+	var/mob/abstract/eye/component_eye
```

Comment:
Null or QDEL_NULL this reference on Destroy

Reply:
It's QDEL_NULLed on unlook which is called on Destroy